src/azure-account.ts

```diff
@@ -376,9 +380,21 @@ export class AzureLoginHelper {
 		if (selected) {
 			const config = workspace.getConfiguration('azure');
 			if (config.get('cloud') !== selected.environment.name) {
+				if (selected.environment.name === azureCustomCloud) {
+					const armUrl = await window.showInputBox({
+						prompt: localize('azure-account.enterArmUrl', "Enter the Azure Resource Manager endpoint"),
+						placeHolder: 'https://management.local.azurestack.external'
+					});
+					if (armUrl) {
+						await config.update(customCloudArmUrlKey, armUrl, getCurrentTarget(config.inspect(customCloudArmUrlKey)));
```

Comment:
There is an `onDidChangeConfiguration` listener earlier in this file that need to check if this key has changed

Reply:
`this.doLogin = true;` also needs to be set here, before the update